libnopegl/src/node_drawother.c

```diff
@@ -1208,74 +1049,81 @@ static int drawmask_prepare(struct ngl_node *node)
+    return finalize_init(node, &s->common, &crafter_params);
 }
 
-static int drawnoise_prepare(struct ngl_node *node)
+static int drawnoise_init(struct ngl_node *node)
 {
     struct drawnoise_priv *s = node->priv_data;
     struct drawnoise_opts *o = node->opts;
+    if (o->octaves < 1 || o->octaves > 8) {
+        LOG(ERROR, "octaves must be in [1,8]");
+        return NGL_ERROR_INVALID_ARG;
+    }
 
-    struct pipeline_desc *desc = create_desc(node);
-    if (!desc)
-        return NGL_ERROR_MEMORY;
+    s->common.helpers = NGLI_FILTER_HELPER_MISC_UTILS | NGLI_FILTER_HELPER_NOISE;
+    int ret = init(node, &s->common, &o->common, "source_noise", source_noise_frag);
+    if (ret < 0)
+        return ret;
 
     const struct pgcraft_uniform uniforms[] = {
-        {.name="type",              .type=NGPU_TYPE_I32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=&o->type},
-        {.name="amplitude",         .type=NGPU_TYPE_F32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->amplitude_node, &o->amplitude)},
-        {.name="octaves",           .type=NGPU_TYPE_U32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=&o->octaves},
-        {.name="lacunarity",        .type=NGPU_TYPE_F32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->lacunarity_node, &o->lacunarity)},
-        {.name="gain",              .type=NGPU_TYPE_F32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->gain_node, &o->gain)},
-        {.name="seed",              .type=NGPU_TYPE_U32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->seed_node, &o->seed)},
-        {.name="scale",             .type=NGPU_TYPE_VEC2, .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->scale_node, o->scale)},
-        {.name="evolution",         .type=NGPU_TYPE_F32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->evolution_node, &o->evolution)},
+        {.name="type",       .type=NGPU_TYPE_I32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=&o->type},
+        {.name="amplitude",  .type=NGPU_TYPE_F32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->amplitude_node, &o->amplitude)},
+        {.name="octaves",    .type=NGPU_TYPE_U32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=&o->octaves},
+        {.name="lacunarity", .type=NGPU_TYPE_F32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->lacunarity_node, &o->lacunarity)},
+        {.name="gain",       .type=NGPU_TYPE_F32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->gain_node, &o->gain)},
+        {.name="seed",       .type=NGPU_TYPE_U32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->seed_node, &o->seed)},
+        {.name="scale",      .type=NGPU_TYPE_VEC2, .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->scale_node, o->scale)},
+        {.name="evolution",  .type=NGPU_TYPE_F32,  .stage=NGPU_PROGRAM_SHADER_FRAG, .data=ngli_node_get_data_ptr(o->evolution_node, &o->evolution)},
```

Comment:
These commits are large, can you avoid adding more cosmetics to them? 😅

Reply:
Removed.